### PR TITLE
Upstream shape function for embedding_bag.padding_idx from Torch-MLIR

### PR DESCRIPTION
**Pull request by vidsinghal**
### Description
 Adding upstream shape function for embedding_bag.padding_idx

### Issue
 None

### Testing
 None

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

**Review by Gamrix — APPROVED** [review]
Looks good, I have released this diff to the testing flow, and happy to land when test pass

File: torch/jit/_shape_functions.py [hunk 1]
```diff
@@ -439,6 +439,34 @@ def embedding(
     size.append(weight[1])
     return size
 
+def embedding_bag_padding_idx(weight: List[int], indices: List[int], offsets: List[int], scale_grad_by_freq: bool, mode: int, sparse: int, per_sample_weights: Optional[List[int]], include_last_offset: bool, padding_idx: Optional[int]) -> Tuple[List[int], List[int], List[int], List[int]]):
```
Comment (Gamrix) [inline torch/jit/_shape_functions.py:442]: You do not need to include the arguments that are not used inside the shape function.

**Review by Gamrix — CHANGES_REQUESTED** [review]
Actually, I realized that the shape function test will not pass until you generated the cpp definitions like here https://github.com/pytorch/pytorch/pull/79194/files. you need to run [gen_jit_shape_functions.py](https://github.com/pytorch/pytorch/blob/45cb8199df131a333615123e76c9cf5b37b7cdf1/torchgen/shape_functions/gen_jit_shape_functions.py#L14-L17).

File: torch/jit/_shape_functions.py (hunk 1 above)
Comment (vid-999) [reply torch/jit/_shape_functions.py]: makes sense.

**Comment by vid-999** [comment]
> Actually, I realized that the shape function test will not pass until you generated the cpp definitions like here https://github.com/pytorch/pytorch/pull/79194/files. you need to run [gen_jit_shape_functions.py](https://github.com/pytorch/pytorch/blob/45cb8199df131a333615123e76c9cf5b37b7cdf1/torchgen/shape_functions/gen_jit_shape_functions.py#L14-L17).

Completed.

**Comment by vid-999** [comment]
> Looks good, I have released this diff to the testing flow, and happy to land when test pass

thanks!

**Comment by vid-999** [comment]
The second argument shape was incorrect i think. I was using mode == 1 when is should have been mode == 0. 
corrected that.

**Review by Gamrix — APPROVED** [review]
Looks good

**Comment by Gamrix** [comment]
Can you rebase on `viable/strict`, and rerun `gen_jit_shape_functions.py`? The lintrunner error is something that I fixed a few days ago.

**Comment by vid-999** [comment]
> Can you rebase on `viable/strict`, and rerun `gen_jit_shape_functions.py`? The lintrunner error is something that I fixed a few days ago. You will hit merge conflicts with `serialized_shape_function_registry.cpp` in your rebase, but you can ignore them as rerunning `gen_jit_shape_functions.py` will fix the merge conflicts.

Hi, I have rebase on top of viable/strict and re-run serialized_shap_function_registry.cpp

File: torch/csrc/jit/runtime/serialized_shape_function_registry.cpp [hunk 2]
```diff
@@ -421,8 +478,7 @@ def mm(self: List[int],
     ops.prim.RaiseException("AssertionError: ")
   return [self[0], mat2[1]]
 
-)=====")
-+ std::string(R"=====(def dot(self: List[int],
+def dot(self: List[int],
```
Comment (Gamrix) [inline torch/csrc/jit/runtime/serialized_shape_function_registry.cpp:481]: These changes should no longer be necessary due to the this commit I landed:
https://github.com/pytorch/pytorch/pull/83092

If you need to manually edit the generated file for any reason, let me know, and I can look into a fix for it.

**Review by Gamrix — CHANGES_REQUESTED** [review]
I don't think there should be any more manual changes needed for the generated file. The current master passes tests as-is due to some recent fixes that I landed. If you are not find that the case, let me know.

File: torch/csrc/jit/runtime/serialized_shape_function_registry.cpp (hunk 2 above)
Comment (vid-999) [reply torch/csrc/jit/runtime/serialized_shape_function_registry.cpp:481]: Hi I rebased on top of my branch using the following command
```
git pull --rebase upstream viable/strict
```
re-ran gen_jit_shape_functions.py but I don't see any changes to the serialized_shape_function_registry.cpp file.

Comment (Gamrix) [reply torch/csrc/jit/runtime/serialized_shape_function_registry.cpp:481]: Did you rebuild Pytorch first?

Comment (Gamrix) [reply torch/csrc/jit/runtime/serialized_shape_function_registry.cpp:481]: The changes to `sum_mean_dim` for example should no longer be happening. What happens if you delete `serialized_shape_function_registry.cpp` and rerun gen_jit_shape_functions.py?

Comment (vid-999) [reply torch/csrc/jit/runtime/serialized_shape_function_registry.cpp:481]: yes

Comment (vid-999) [reply torch/csrc/jit/runtime/serialized_shape_function_registry.cpp:481]: hi I have re-ran and pushed the changes but the changes to sum_mean_dim still seem to be happening.

**Comment by Gamrix** [comment]
I think I have figured out what has happened, and have put out a fix for it here #83629 , rebasing on this then running the file should fix the issue you are hitting. 
Alternatively, you should be able to run `python setup.py develop` then rerun the script and it should fix itself.

**Comment by vid-999** [comment]
> I think I have figured out what has happened, and have put out a fix for it here #83629 , rebasing on this then running the file should fix the issue you are hitting. Alternatively, you should be able to run `python setup.py develop` then rerun the script and it should fix itself.

Hi thanks for the help, it worked after incorporating the PR you mentioned.

**Pull request closed without merging**